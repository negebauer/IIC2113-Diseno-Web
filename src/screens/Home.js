import React, { Component } from "react"
import { connect } from "react-redux"

import { devlog } from "../utils/log"
import { ContainerCenter } from "../components/Container"

const mapStateToProps = () => ({})

const mapDispatchToProps = {}

class Home extends Component {
  render() {
    devlog("Home", this.props)
    return (
      <ContainerCenter>
        <h1>IIC2113 - Web</h1>
        <h3>
          Api is at url {process.env.REACT_APP_API || "NO_ENV"}
        </h3>
      </ContainerCenter>
    )
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
