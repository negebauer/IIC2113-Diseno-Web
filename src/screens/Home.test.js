// Maybe check
// https://github.com/Gethyl/ReactReduxTestingUsingJestEnzyme/blob/master/__test__/Home.spec.js

import React from 'react'
import configureStore from '../redux/store'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Home from './Home'

let initialState
let store

beforeEach(() => {
  initialState = {}
  store = configureStore(initialState)
})

it('shallow renders without crashing', () => {
  shallow(<Home store={store} />)
})

it('mount renders without crashing', () => {
  mount(<Home store={store} />)
})

it('matches snapshot', () => {
  const wrapper = mount(<Home store={store} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders children', () => {
  const wrapper = mount(<Home store={store} />)
  expect(wrapper.children().exists()).toBeTruthy()
})
