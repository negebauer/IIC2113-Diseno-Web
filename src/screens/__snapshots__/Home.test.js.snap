// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Connect(Home)
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Home
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="container"
    >
      <h1>
        IIC2113 - Web
      </h1>
      <h3>
        Api is at url 
        NO_ENV
      </h3>
    </div>
  </Home>
</Connect(Home)>
`;
